src/frontend/js/modules/session.js

```diff
@@ -92,19 +94,28 @@
     $scope.domain = session.domain;
   })
 
-  .controller('sessionInitESNController', function($scope, esnTemplate, sessionFactory) {
+  .controller('sessionInitESNController', function($scope, esnTemplate, sessionFactory, session, themesService, applyThemeService) {
 
     $scope.session = {
       template: esnTemplate.templates.loading
     };
 
+    // TODO: Refactor this, maybe use async await to avoid nested promises?
     sessionFactory.fetchUser(function(error) {
       if (error) {
         $scope.session.error = error.data;
         $scope.session.template = esnTemplate.templates.error;
-      } else {
-        $scope.session.template = esnTemplate.templates.success;
+        return;
       }
+
+      themesService.getTheme(session.domain._id).then(function(theme) {
```

Comment:
I prefer you add this code in the `fetchUser` service method. Also update the fetchUser service method to have:

```javascript
userAPI.currentUser().then(() => {
  return fetchDomain(...);
}).then(() => {
  return fetchTheme(...).applyTheme(...);
})
.then(() => { success story })
.catch(() => { error story });
```
Maybe "fetchUser" is not a good word also, and the method name should be renamed to something like "bootstrapSession()". I let you see this.

Reply:
Point's taken. I'll adjust it accordingly.

Reply:
I have just double-checked before changing the `fetchUser` function. There are 2 other places where it is used:

- https://github.com/OpenPaaS-Suite/esn-frontend-mailto/blob/93703949be8829856244daf7ef3af9247f7ca3e4/src/app/mailto.run.js#L7
- https://github.com/OpenPaaS-Suite/esn-frontend-inbox/blob/48432fe893a1ae0c78bf713ce5781129afbdf2e0/src/linagora.esn.unifiedinbox/app/mailto/mailto.run.js#L7

Btw, I'm not sure why the same piece of code above (`mailto.run.js`) is duplicated in 2 different modules.

Reply:
@chamerling you created the mailto app: can you confirm that we can now safely remove the "mailto" related code from esn-frontend-inbox ? If yes, can you create an issue in the **Split Pilish** project for this ?

Reply:
@MichaelBailly yes tue, I will create issue and remove it

Reply:
@chamerling can we have more focus please ? I asked the issue being created in the Split Polish backlog, even put it in **bold** , and is now done and in the Frontend Split board... https://github.com/OpenPaaS-Suite/esn-frontend-inbox/pull/15 What was not clear in my instructions ? I'm OK to add extra tasks to the current project, but once we have completed it. Which is not the case as of today.

Reply:
OMG, I did not read the project name correctly, created in the Split one and fixed it in 2 minutes :p